pkg/products/observability/reconciler_test.go

```diff
@@ -542,8 +542,9 @@ func TestReconciler_fullReconcile(t *testing.T) {
 func basicInstallation() *v1alpha1.RHMI {
 	return &v1alpha1.RHMI{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "installation",
-			Namespace: defaultInstallationNamespace,
+			Name:            "installation",
+			Namespace:       defaultInstallationNamespace,
+			ResourceVersion: "1",
```

Comment:
Currently, this `basicInstallation()` function is being using in test case with only one test case. As the function returns a pointer setting the `ResourceVersion` will cause an issue. The fact a pointer is returned is the root of the problem.

Reply:
As verified on call together, the issue mentioned here is not present in the tests that are refactored.